File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        res=[]
        i,j=0,1
        if len(intervals)==0 or len(intervals)==1:
            return intervals
        while j<len(intervals):
            if max(intervals[i])>=min(intervals[j]):
                res.append([min(intervals[i]),max(intervals[j])])
                i=j+1
                j+=1
            else:
                res.append(intervals[i])
                i=j
                continue
            j+=1
        return res
        

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a,b) -> a[0] - b[0]);
        int[][] res = new int[intervals.length][2];
        int l=0;
        res[0] = intervals[0];
        for(int i=1;i<intervals.length;i++){
            if(res[l][1]>=intervals[i][0]){
                res[l][1] = Math.max(res[l][1],intervals[i][1]);
            }
            else{
                l++;
                res[l] = intervals[i];
            }
            
        }
        return Arrays.copyOf(res,l+1);
    }
}